src/mbgl/map/map_context.cpp

```diff
@@ -112,7 +112,12 @@ void MapContext::setStyleURL(const std::string& url) {
         if (res.status == Response::Successful) {
             loadStyleJSON(res.data, base);
         } else {
-            Log::Error(Event::Setup, "loading style failed: %s", res.message.c_str());
+            if (res.message.substr(res.message.length() - 3) == "404" &&
```

Comment:
The way I see to share this code would be create something like `mbgl::util::HTTPResponseStatusCodeToString(uint32_t)` and call it from the `http_request_*`.

But IMO it is also OK do it by adding an `else if` on both `http_request_nsurl.mm` and `http_request_curl.cpp` and duplicate the string with the error message, which is how we do it today.